src/pages/index.js

```diff
@@ -95,5 +156,48 @@ const validateAddPlace = new FormValidator(config, addPlaceForm);
 validateAddPlace.enableValidation();
 
 const popupWithImage = new PopupWithImage({ popupSelector: "#image-modal" });
-
 popupWithImage.setEventListeners();
+
+// Avatar
+const updateAvatarModal = document.querySelector("#update-avatar");
+const avatarForm = updateAvatarModal.querySelector(".modal__form");
+const updateAvatarButton = document.querySelector(".profile__image_edit-icon");
+
+const validateUpdateAvatar = new FormValidator(config, avatarForm);
+
+const popupUpdateAvatar = new PopupWithForm(
+  "#update-avatar",
+  handleUpdateAvatar
+);
+popupUpdateAvatar.setEventListeners();
+
+function handleUpdateAvatar({ link }) {
+  const saveButton = document.querySelector("#avatar-submit-button");
+  const originalText = saveButton.textContent;
+
+  saveButton.textContent = "Saving...";
+  api
+    .updateAvatar(link)
+    .then((data) => {
+      userInfo.setUserInfo({ avatar: data.avatar });
+      popupUpdateAvatar.close();
+      avatarForm.reset();
+      validateUpdateAvatar.toggleButtonState();
+    })
+    .catch((err) => {
+      console.error(err);
+    }).finally(()=> {
+      saveButton.textContent = originalText;
+    });
+}
+
+updateAvatarButton.addEventListener("click", () => {
+  popupUpdateAvatar.open();
+  validateUpdateAvatar.enableValidation();
```

Comment:
Needs corection: the validation should be enabled only once in the global scope (please fix that for all the forms if needed)

Reply:
Not sure how I would do this, I have been setting validation for each form separately

Reply:
Yes, but you just should do that in the global scope as the validation should be enabled only once, not every time you open the form:

```js
validateUpdateAvatar.enableValidation();

updateAvatarButton.addEventListener("click", () => {
  popupUpdateAvatar.open();
})
```